### PR TITLE
Local Storage

### PR DESCRIPTION
**Pull request by gergitokusse**
# The following file was Updated

- Jscript.js was updated for local storage feature

### CODE REVIEW

**Review by Ibizugbe — COMMENTED** [review]
Hi @gergitokusse 👋👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that lacks contribution from your learning partner🥲 don't forget that this is a pair programming activity and the contributions of each partenr is necessary👍 Kindly reach out to your learning partner and work with him👍

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: jscript.js [hunk 1]
```diff
@@ -389,6 +393,10 @@ form.addEventListener('submit', (event) => {
     showError();
     event.preventDefault();
   }
+  // local storage
+  const mylocaldata = { Name: nameval.value, Email: email.value, Message: messageval.value };
+  const sdata = JSON.stringify(mylocaldata);
+  localStorage.setItem('myLocalData', sdata);
```
Comment (AbelG101) [inline jscript.js]: - [ ] Ouch! this is so close, but as specified in the project requirements you will have to save the form data in the `local storage`  when the user changes the content of **any input field** and not when the form is submitted so kindly make sure you fix that. ✔

File: jscript.js [hunk 2]
```diff
@@ -323,7 +323,11 @@ const messageval = document.getElementById('message');
 const emailError = document.querySelector('.erroremail');
 const nameerror = document.querySelector('.errorname');
 const messageerror = document.querySelector('.errormessage');
-
+// parsing data from local storage
+const storage = JSON.parse(localStorage.getItem('myLocalData'));
+nameval.value = storage.Name;
+email.value = storage.Email;
+messageval.value = storage.Message;
```
Comment (AbelG101) [inline jscript.js]: - [ ]  Please make sure to check whether the local storage is empty or not before setting the input values as it currently throws an error in the console when accessing a null value.
  image
   - You should add an `if` statement to check  if the local storage is empty before setting the values 😉

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @gergitokusse & @techmoves 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Correct use of GitHub Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- You just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: jscript.js (hunk 1 above)
Comment (gergitokusse) [reply jscript.js]: # 👉 Okay I will do that and will fix it !!!

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @gergitokusse 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer
- Excellent communication 

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**